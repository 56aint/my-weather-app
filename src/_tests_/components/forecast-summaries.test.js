import React from 'react';
import { cleanup, render } from '@testing-library/react';
import ForecastSummaries from '../../components/forecast-summaries';
// import forecast from '../../data/forecast.json';

/*const forecasts = [
  {
    date: 123,
    description: 'date',
    icon: 'icon1',
    temperature: {
      max: 999,
    },
  },
  {
    date: 456,
    description: 'date2',
    icon: 'icon2',
    temperature: {
      max: 777,
    },
  }
];*/

afterEach(cleanup);

describe("ForecastSummaries", () => {
  const forecasts = [
    {
      date: 123,
      description: 'date',
      icon: 'icon1',
      temperature: {
        max: 999,
      }
    },
    {
      date: 456,
      description: 'date2',
      icon: 'icon2',
      temperature: {
        max: 777,
      },
    },
  ];

  it("renders correctly to match ForcastSummaries snapshot", () => {
    const { asFragment } = render(<ForecastSummaries forecasts={forecasts} />);

    expect(asFragment).toMatchSnapshot();
  });
  
  it("renders the correct amount of ForecastSummary props", () => {
    const { getAllByTestId } = render(<ForecastSummaries forecasts={forecasts} />)

    expect(getAllByTestId("date-id")).toHaveLength(2);
    expect(getAllByTestId("temperature-id")).toHaveLength(2);
    expect(getAllByTestId("description-id")).toHaveLength(2);
    expect(getAllByTestId("icon-id")).toHaveLength(2);
  });
});