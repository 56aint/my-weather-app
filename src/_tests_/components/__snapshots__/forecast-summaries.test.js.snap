// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly to match ForcastSummaries snapshot 1`] = `[Function]`;
